### PR TITLE
Adds baystation-style volumetric space storage support

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request
image

## Why It's Good For The Game

Allows for more granular control over how much people can carry to rebalance things with instead of the inflexible one-dimensional w-class system.

## Changelog
:cl:
add: Volumetric storage is here.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
are you able to click on the backround of the items? - Like how you could before
image
I.e if I click were the blue is, will I pick up the glass?

**Comment by silicons** [comment]
@Trilbyspaceclone yeah the background boxes direct clicks to the items

 i had to rewrite baystorage (otherwise i could have just ported but i wouldn't because i don't like baycode lol) but it was well worth it, heck baystation's no-click boxes, mine are clickable :^)

File: code/datums/components/storage/ui.dm [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	return TRUE
+
+/**
+  * Returns TRUE if we are using volumetric UI instead of box UI
+  */
+/datum/component/storage/proc/volumetric_ui()
+	var/atom/real_location = real_location()
+	return (storage_flags & STORAGE_LIMIT_VOLUME) && (length(real_location.contents) <= MAXIMUM_VOLUMETRIC_ITEMS) && !display_numerical_stacking
+
+/**
```
Comment (silicons) [inline code/datums/components/storage/ui.dm:249]: anyone wondering why i'm spamming getters:

yeah it's pretty terrible to have every storage component making 2 screen objects much less 5 even if no player uses it

**Review by Ghommie — APPROVED** [review]
Time to crash test it on the live servers.

**Pull request merged**